Documentation/generics.md

```diff
@@ -0,0 +1,64 @@
+*.NET Innovation Examples*
+**GENERICS**
+===================
+
+We use generics implicitly all the time in  C#. When you use LINQ in C#, do you notice you are working with <code>IEnumerable\<T\></code>? Ever wonder what <code>T</code> is and what it means?
```

Comment:
There are plenty of explicit uses of generics, too. List<T> and Dictionary<K,V> have to be among the most popular .NET classes. Also, people do use IEnumberable<T> directly, too.

I'd lead more with the explicit cases of generics being useful and then follow with the implicit cases. 

Before generics, everyone was using ArrayList and then casting all the time. That requirement is now gone.